kolibri/core/content/models.py

```diff
@@ -98,6 +98,78 @@ def get_queryset(self, *args, **kwargs):
             .order_by(self.tree_id_attr, self.left_attr)
         )
 
+    def build_tree_nodes(self, data, target=None, position="last-child"):
+        """
+        vendored from:
+        https://github.com/django-mptt/django-mptt/blob/fe2b9cc8cfd8f4b764d294747dba2758147712eb/mptt/managers.py#L614
+
+        Load a tree from a nested dictionary for bulk insert, returning an
+        array of records. Use to efficiently insert many nodes within a tree
+        without an expensive `rebuild`.
+        ::
+            records = MyModel.objects.build_tree_nodes({
```

Comment:
The `docs` TOX env doesn't like something about the code indentation around here

Reply:
Yeah - I just removed it as it's been vendored and has  a link back.